### PR TITLE
Rewards page scaffold

### PR DESCRIPTION
**Pull request by masvelio**
## Link to Issue
Closes: #10325 

## Description of Changes

1. **New Rewards Page Feature**
- Created a new `/rewards` route accessible in both common and custom domains
- Implemented a new rewards page with three main sections:
  - Referrals
  - Wallet Balance
  - Quests

2. **UI Components**
- Added new components:
  - `RewardsCard` - A reusable card component for the rewards page
  - `RewardsTab` - Mobile tab navigation component
  - Table components for each section (Referral, Quest, Wallet)

3. **Navigation Updates**
- Added rewards button to desktop header with ETH balance display
- Added rewards tab to mobile navigation when feature is enabled
- Updated "My transactions" link to redirect to rewards page when enabled

4. **Feature Flag Integration**
- Added `FLAG_REWARDS_PAGE` environment variable
- Integrated feature flag checks throughout the application

## Test Plan
- FLAG_REWARDS_PAGE to true 
- check /rewards page
- only UI for now

image

image

## Deployment Plan

1. n/a

## Other Considerations

- n/a

### CODE REVIEW

File: packages/commonwealth/client/scripts/helpers/feature-flags.ts [hunk 1]
```diff
@@ -32,6 +32,7 @@ const featureFlags = {
   referrals: buildFlag(process.env.FLAG_REFERRALS),
   stickyEditor: buildFlag(process.env.FLAG_STICKY_EDITOR),
   newMobileNav: buildFlag(process.env.FLAG_NEW_MOBILE_NAV),
+  rewardsPage: buildFlag(process.env.FLAG_REWARDS_PAGE),
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/helpers/feature-flags.ts:35]: will this be launched as a separate feature or will be needed with both xp/quests as each of those are rolled out?

File: packages/commonwealth/client/scripts/views/components/SublayoutHeader/useUserMenuItems.tsx [hunk 2]
```diff
@@ -297,7 +299,14 @@ const useUserMenuItems = ({
       {
         type: 'default',
         label: 'My transactions',
-        onClick: () => navigate(`/myTransactions`, {}, null),
+        onClick: () =>
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/SublayoutHeader/useUserMenuItems.tsx:302]: shouldn't the label be updated to "My rewards" when `rewardsEnabled=true`?

File: packages/commonwealth/client/scripts/views/pages/RewardsPage/utils.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
+export enum MobileTabType {
+  Referrals = 'Refferal Earnings',
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/RewardsPage/utils.ts:2]: typo

File: packages/commonwealth/client/scripts/views/pages/RewardsPage/utils.ts
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/RewardsPage/utils.ts:1]: maybe move the types to types.ts and keep the mappings here

**Review by mzparacha — APPROVED** [review]

File: packages/commonwealth/client/scripts/views/components/SublayoutHeader/useUserMenuItems.tsx (hunk 2 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/components/SublayoutHeader/useUserMenuItems.tsx:302]: nope, this label is adequate, it takes user to specific tab/table in the /rewards page

File: packages/commonwealth/client/scripts/helpers/feature-flags.ts (hunk 1 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/helpers/feature-flags.ts:35]: for now this is separate flag but this should be adjusted to be accessible whenever the first feature goes out (referrals/quests)

File: packages/commonwealth/client/scripts/views/pages/RewardsPage/utils.ts (hunk 3 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/RewardsPage/utils.ts:2]: thanks, actually i had more typos so went through all of them

File: packages/commonwealth/client/scripts/views/pages/RewardsPage/utils.ts
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/RewardsPage/utils.ts:1]: done

**Review by Israellund — APPROVED** [review]

**Pull request merged**